pkg/virt-operator/controller/kubevirt.go

```diff
@@ -0,0 +1,155 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+
+package controller
+
+import (
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/controller"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/pkg/log"
+)
+
+type KubeVirtController struct {
+	clientset        kubecli.KubevirtClient
+	queue            workqueue.RateLimitingInterface
+	kubeVirtInformer cache.SharedIndexInformer
+	recorder         record.EventRecorder
+}
+
+func NewKubeVirtController(
+	clientset kubecli.KubevirtClient,
+	informer cache.SharedIndexInformer,
+	recorder record.EventRecorder) *KubeVirtController {
+
+	c := KubeVirtController{
+		clientset:        clientset,
+		queue:            workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
+		kubeVirtInformer: informer,
+		recorder:         recorder,
+	}
+
+	c.kubeVirtInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    c.addKubeVirt,
+		DeleteFunc: c.deleteKubeVirt,
+		UpdateFunc: c.updateKubeVirt,
+	})
+
+	return &c
+}
+
+func (c *KubeVirtController) addKubeVirt(obj interface{}) {
+	c.enqueueKubeVirt(obj)
+}
+
+func (c *KubeVirtController) deleteKubeVirt(obj interface{}) {
+	c.enqueueKubeVirt(obj)
+}
+
+func (c *KubeVirtController) updateKubeVirt(old, curr interface{}) {
```

Comment:
why do you have 'old' interface if it is not being used ?

Reply:
It implements an interface.

Reply:
OK